lambdas/tests/unit/handlers/test_authoriser_handler.py

```diff
@@ -1,214 +1,291 @@
-#     expired_session = {
-#         "Count": 1,
-#         "Items": [
-#             {
-#                 "NDRSessionId": "test_session_id",
-#                 "sid": "test_cis2_session_id",
-#                 "Subject": "test_cis2_user_id",
-#                 "TimeToExist": one_minute_ago,
-#             }
-#         ],
-#     }
-#     mock_session_table.query.return_value = expired_session
-
-#     test_event = {
-#         "authorizationToken": "valid_pcse_token",
-#         "methodArn": f"{MOCK_METHOD_ARN_PREFIX}/GET/SearchPatient",
-#     }
-
-#     response = lambda_handler(test_event, context=None)
-
-#     assert response["policyDocument"] == DENY_ALL_POLICY
-
-
-# def test_invalid_token_return_deny_policy(mocker, mock_ssm, mock_session_table):
-#     decode_mock = mocker.patch(
-#         "jwt.decode", side_effect=jwt.exceptions.InvalidTokenError
-#     )
-
-#     test_event = {
-#         "authorizationToken": "invalid_token",
-#         "methodArn": f"{MOCK_METHOD_ARN_PREFIX}/GET/SearchPatient",
-#     }
-#     response = lambda_handler(test_event, context=None)
-
-#     decode_mock.assert_called_with(
-#         "invalid_token", TEST_PUBLIC_KEY, algorithms=["RS256"]
-#     )
-#     assert response["policyDocument"] == DENY_ALL_POLICY
-
-
-# def test_invalid_signature_return_deny_policy(mocker, mock_ssm, mock_session_table):
-#     decode_mock = mocker.patch(
-#         "jwt.decode", side_effect=jwt.exceptions.InvalidSignatureError
-#     )
-
-#     test_event = {
-#         "authorizationToken": "token_with_invalid_signature",
-#         "methodArn": f"{MOCK_METHOD_ARN_PREFIX}/GET/SearchPatient",
-#     }
-#     response = lambda_handler(test_event, context=None)
-
-#     decode_mock.assert_called_with(
-#         "token_with_invalid_signature", TEST_PUBLIC_KEY, algorithms=["RS256"]
-#     )
-
-#     assert response["policyDocument"] == DENY_ALL_POLICY
+import os
+import time
+from unittest.mock import patch
+
+import jwt
+import pytest
+from enums.repository_role import RepositoryRole
+from handlers.authoriser_handler import lambda_handler, validate_access_policy
+
+MOCK_METHOD_ARN_PREFIX = "arn:aws:execute-api:eu-west-2:fake_arn:fake_api_endpoint/dev"
+TEST_PUBLIC_KEY = "test_public_key"
+DENY_ALL_POLICY = {
+    "Statement": [
+        {
+            "Action": "execute-api:Invoke",
+            "Effect": "Deny",
+            "Resource": [f"{MOCK_METHOD_ARN_PREFIX}/*/*"],
+        }
+    ],
+    "Version": "2012-10-17",
+}
+
+
+@pytest.fixture
+def patch_env_vars():
+    env_vars = {
+        "SSM_PARAM_JWT_TOKEN_PUBLIC_KEY": "test_jwt_token_public_key",
+        "AUTH_SESSION_TABLE_NAME": "test_session_table",
+    }
+    with patch.dict(os.environ, env_vars):
+        yield env_vars
+
+
+@pytest.fixture()
+def mock_ssm(patch_env_vars, mocker):
+    mock_ssm_client = mocker.patch("boto3.client")
+    mock_ssm_client.return_value.get_parameter.return_value = {
+        "Parameter": {"Value": TEST_PUBLIC_KEY}
+    }
```

Comment:
Can you patch our ssm service instead of boto3?

Reply:
done